src/tests/Metalama.Patterns.Observability.AspectTests/WriteabilityInpc.t.cs

```diff
@@ -0,0 +1,413 @@
+public class Person : INotifyPropertyChanged
+{
+  private string? _firstName;
+  public string? FirstName
+  {
+    get
+    {
+      return _firstName;
+    }
+    set
+    {
+      if (!object.ReferenceEquals(value, _firstName))
+      {
+        _firstName = value;
+        OnPropertyChanged("FirstName");
+      }
+    }
+  }
+  private string? _lastName;
+  public string? LastName
+  {
+    get
+    {
+      return _lastName;
+    }
+    set
+    {
+      if (!object.ReferenceEquals(value, _lastName))
+      {
+        _lastName = value;
+        OnPropertyChanged("LastName");
+      }
+    }
+  }
+  protected virtual void OnPropertyChanged(string propertyName)
+  {
+    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+  }
+  public event PropertyChangedEventHandler? PropertyChanged;
+}
+[Observable]
+public class PersonViewModelField : INotifyPropertyChanged
+{
+  private Person _model1 = default !;
+  private Person _model
+  {
+    get
+    {
+      return _model1;
+    }
+    set
+    {
+      if (!object.ReferenceEquals(value, _model1))
+      {
+        var oldValue = _model1;
+        if (oldValue != null)
+        {
+          oldValue.PropertyChanged -= _handle_modelPropertyChanged;
+        }
+        _model1 = value;
+        OnPropertyChanged("FirstName");
+        OnPropertyChanged("FullName");
+        OnPropertyChanged("LastName");
+        SubscribeTo_model(value);
+      }
+    }
+  }
+  public PersonViewModelField(Person model)
+  {
+    this._model = model;
+  }
+  public string? FirstName => this._model.FirstName;
+  public string? LastName => this._model.LastName;
+  public string FullName => $"{this.FirstName} {this.LastName}";
+  private PropertyChangedEventHandler? _handle_modelPropertyChanged;
+  protected virtual void OnPropertyChanged(string propertyName)
+  {
+    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+  }
+  private void SubscribeTo_model(Person value)
+  {
+    if (value != null)
+    {
+      _handle_modelPropertyChanged ??= HandlePropertyChanged;
+      value.PropertyChanged += _handle_modelPropertyChanged;
+    }
+    void HandlePropertyChanged(object? sender, PropertyChangedEventArgs e)
+    {
+      {
+        var propertyName = e.PropertyName;
+        switch (propertyName)
+        {
+          case "FirstName":
+            OnPropertyChanged("FirstName");
+            OnPropertyChanged("FullName");
+            break;
+          case "LastName":
+            OnPropertyChanged("FullName");
+            OnPropertyChanged("LastName");
+            break;
+        }
+      }
+    }
+  }
+  public event PropertyChangedEventHandler? PropertyChanged;
+}
+[Observable]
+public class PersonViewModelReadonlyField : INotifyPropertyChanged
+{
+  private readonly Person _model1 = default !;
+  private Person _model
+  {
+    get
+    {
+      return _model1;
+    }
+    init
+    {
+      if (!object.ReferenceEquals(value, _model1))
+      {
```

Comment:
The OnPropertyChanged code must not be used here. Only the call to SubscribeTo_model.

Reply:
A better strategy would be to call SubscribeTo from the constructor as an initializer.